chore(op-acceptor): clean; remove unused code identified by static analysis tools

scharissis opened the pull request:
- Used deadcode and staticcheck to identify 11 unused functions
- Removed ~600 lines of code across multiple packages

scharissis committed:
chore(op-acceptor): clean; remove unused code identified by static analysis tools

- Used deadcode and staticcheck to identify 11 unused functions
- Removed ~600 lines of code across multiple packages

janjakubnanista commented on op-acceptor/cmd/main_test.go:
```diff
@@ -202,7 +202,7 @@ gates:
 			tempDir, err := os.MkdirTemp("/tmp", "op-acceptor-test-")
 			require.NoError(t, err, "Failed to create temporary directory")
-			defer os.RemoveAll(tempDir)
+			defer func() { _ = os.RemoveAll(tempDir) }()
```
Why this change? Not that I am a go expert but this seems like just introducing a new lambda

scharissis replied:
To appease the linter. It was complaining that we weren't checking the error code which [os.RemoveAll](https://pkg.go.dev/os#RemoveAll) returns.

janjakubnanista commented on op-acceptor/logging/filelogger_test.go:
```diff
@@ -714,93 +714,6 @@ func TestHTMLSummarySink_WithSubtestsAndNetworkInfo(t *testing.T) {
 // TestExtractErrorInfoFromJSON verifies that error information is correctly extracted from test output
 func TestExtractErrorInfoFromJSON(t *testing.T) {
```
This seems like it left an empty test function behind

janjakubnanista replied:
Oh and the `extractErrorData` is still there - we sure we want to remove the tests?

scharissis replied:
Good spot, it's indeed still used, so I'll restore this test. Cheers

scharissis committed:
refactor(templates): centralize template functions and resolve circular dependencies

- Create new templates package with centralized template functions
- Remove duplicate template function definitions from logging and reporting packages
- Resolve circular dependency issues between packages
- Fix linter issues with error handling patterns (type assertions -> errors.As)
- Restore test coverage for extractErrorData function that was incorrectly removed
